File: go/clock/clock.go
package clock

import "fmt"

const testVersion = 3

type Clock int

func New(hour, minute int) Clock {
	var totalMinutes = hour%24*60 + minute

	totalMinutes = normalizeMinutes(totalMinutes)

	return Clock(totalMinutes)
}

func (c Clock) String() string {
	var hours = c / 60
	var minutes = c % 60
	return fmt.Sprintf("%02d:%02d", hours, minutes)
}

func (c Clock) Add(minutes int) Clock {
	var sum = int(c) + minutes
	return New(0, sum)
}

func normalizeMinutes(minutes int) int {
	var minutesInDay = 24 * 60

	if minutes < 0 {
		minutes = minutesInDay + minutes
	}

	if minutes > minutesInDay {
		minutes = minutes - minutesInDay
	}

	return minutes
}

File: csharp/clock/solution/solution/Clock.cs
public class Clock
{
    private int hours;
    private int minutes;

    public Clock(int hours)
    {
        this.hours = hours;
    }

    public Clock(int hours, int minutes) : this(hours)
    {
        this.minutes = minutes;
    }

    public Clock Add(int minutes) => new Clock(this.hours, this.minutes + minutes);

    public object Subtract(int minutes) => new Clock(this.hours, this.minutes - minutes);

    public override string ToString()
    {
        while (this.minutes > 59)
        {
            this.minutes -= 60;
            this.hours++;
        }

        while (this.hours > 23)
        {
            this.hours -= 24;
        }

        while (this.minutes < 0)
        {
            this.minutes += 60;
            this.hours--;
        }

        while (this.hours < 0)
        {
            this.hours += 24;
        }

        return $"{hours:d2}:{minutes:d2}";
    }

    public override bool Equals(object obj) => this.ToString() == obj?.ToString();
}